modules/post/windows/gather/enum_hyperv_vms.rb

```diff
@@ -30,11 +35,11 @@ def run
     unless have_powershell?
       fail_with(Failure::NoAccess, "The target does not have PowerShell installed so we can't access the state of the Hyper-V VMs")
     end
-    results = psh_exec('Get-VM')
-    if results =~ /is not recognized as the name of a cmdlet/
-      print_error('The target is not a Hyper-V host')
-    elsif results =~ /do not have the required permission/
-      print_error('You need to be running as an elevated admin or a user of the Hyper-V Administrators group to run this module')
+    get_vm = "try { Get-VM } catch {echo #{error_token}; echo $Error[0]}"
+    results = psh_exec(get_vm)
+    if results =~ /#{error_token}/
+      results.delete!(error_token)
```

Comment:
When I defined `error_token` on my system, this then started to seem a little off. First if the error condition branch is taken, the string will start with the error token, so we can make this a little more specific. Next we can leverage Ruby's `String#delete_prefix` method to cleanly remove just the error token prefix.

```suggestion
    if results.starts_with?(error_token)
      results = results.delete_prefix(error_token).strip
```

Reply:
Committed the changes!